### PR TITLE
Restart-song-test refactor

### PR DESCRIPTION
**Pull request by RenBabiuch**

### CODE REVIEW

File: tests/PageObjects/GamePage.ts [hunk 1]
```diff
@@ -9,4 +9,24 @@ export class GamePagePO {
...
+
+  public get playersScoreElement() {
+    return this.page.getByTestId('players-score');
+  }
+
+  public get currentPlayersScore() {
+    return this.playersScoreElement.getAttribute('data-score');
+  }
+
+  public async waitForPlayersScoreToBeGreaterThan() {
```
Comment (Asvarox) [inline tests/PageObjects/GamePage.ts:25]: It should expect an argument to specify greater than what value the score should be
```
waitForPlayersScoreToBeGreaterThan(expected: number)
```

File: tests/restart-song.spec.ts [hunk 2]
```diff
@@ -11,31 +10,23 @@ test.beforeEach(async ({ page, context, browser }) => {
...
+  await pages.songPreviewPage.goNext();
+  await pages.songPreviewPage.playTheSong();
+  await pages.gamePage.waitForPlayersScoreToBeGreaterThan();
 
   await page.keyboard.press('Backspace');
-  await page.getByTestId('button-restart-song').click();
-
-  await expect(page.getByTestId('players-score')).toHaveAttribute('data-score', '0', { timeout: 15_000 });
+  await pages.gamePage.restartSong();
+  await expect(pages.gamePage.playersScoreElement).toHaveAttribute('data-score', '0', { timeout: 15_000 });
```
Comment (Asvarox) [inline tests/restart-song.spec.ts:31]: could we have a new method in the game page PO `epectPlayersScoreToBe` to hide this "to have attribute" detail from the test?

Comment (RenBabiuch) [reply tests/restart-song.spec.ts:31]: Sure, done. Thanks :)

File: tests/PageObjects/GamePage.ts [hunk 3]
```diff
@@ -9,4 +9,28 @@ export class GamePagePO {
...
+
+  public async waitForPlayersScoreToBeGreaterThan(expected: number) {
+    await expect(async () => {
+      const p1score = await this.currentPlayersScore;
+
+      expect(parseInt(p1score!, 10)).toBeGreaterThan(expected);
+    }).toPass({ timeout: 15_000 });
+  }
+
+  public async expectPlayersScoreToBe() {
```
Comment (Asvarox) [inline tests/PageObjects/GamePage.ts:33]: should also expect an argument `expected: number`, just as the method you fixed above

Comment (RenBabiuch) [reply tests/PageObjects/GamePage.ts]: Aah, right, sorry. Now should be ok

**Review by Asvarox — APPROVED** [review]

**Pull request merged**